runtime/compiler/control/CompilationThread.cpp

```diff
@@ -6269,9 +6352,59 @@ TR::CompilationInfoPerThreadBase::preCompilationTasks(J9VMThread * vmThread,
             *aotCachedMethod = javaVM->sharedClassConfig->findCompiledMethodEx1(vmThread, J9_ROM_METHOD_FROM_RAM_METHOD(method), &flags);
             if (!(flags & J9SHR_AOT_METHOD_FLAG_INVALIDATED))
                {
+#ifdef COMPRESS_AOT_DATA
+               TR_AOTMethodHeader *aotMethodHeader = (TR_AOTMethodHeader*)(((J9JITDataCacheHeader*)*aotCachedMethod) + 1);
+               if (aotMethodHeader->flags & TR_AOTMethodHeader_CompressedMethodInCache)
+                  {
+                  int dataSize = *((uint32_t*)(*aotCachedMethod)-2);
```

Comment:
Why aren't you using `aotMethodHeader->compileMethodDataSize;` ?

Reply:
I see, because `compileMethodDataSize` is the decompressed size. I'm having a hard time finding where you write hte compressed data size.

Reply:
> Could you, as part of this PR, add a .md doc describing the feature. 
I agree with you, let me create a `.md` file describing what feature does and how it works.

Reply:
I guess this gets written in by the SCC library. I think what you should do then is:

```
J9JITDataCacheHeader *dataCacheHeader = (J9JITDataCacheHeader*)*aotCachedMethod;
TR_AOTMethodHeader *aotMethodHeader = (TR_AOTMethodHeader*)(((J9JITDataCacheHeader*)*aotCachedMethod) + 1);
...
int dataSize = dataCacheHeader->size;
```